drivers/adc/adc_stm32.c

```diff
@@ -233,6 +260,86 @@ struct adc_stm32_cfg {
 	size_t pinctrl_len;
 };
 
+#ifdef CONFIG_ADC_STM32_DMA
+static void dma_callback(const struct device *dev, void *arg,
+			 uint32_t channel, int status)
+{
+	/* arg directly holds the adc device */
+	struct adc_stm32_data *data = arg;
+
+	/* invalidate cache to make sure to get the adc data*/
+	SCB_InvalidateDCache_by_Addr(data->buffer, sizeof(uint16_t) * data->channel_count);
+	SCB_InvalidateDCache_by_Addr(data->repeat_buffer, sizeof(uint16_t) * data->channel_count);
+
+	data->samples_count = data->channel_count;
```

Comment:
This line doesn't do anything.
```suggestion
```

Reply:
it keeps the data consistent, so the sample count at the end of conversion should match channel_count